File: php/features.php
<?php
declare(strict_types=1);

// GithubRest SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GithubRestFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GithubRestBaseFeature();
            case "test":
                return new GithubRestTestFeature();
            default:
                return new GithubRestBaseFeature();
        }
    }
}

File: py/features.py
# GithubRest SDK feature factory

from feature.base_feature import GithubRestBaseFeature
from feature.test_feature import GithubRestTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GithubRestBaseFeature(),
        "test": lambda: GithubRestTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
